### PR TITLE
Fix UIPicker on Mac catalyst

### PR DESCRIPTION
**Pull request by annakocheshkova**
Things to consider before you submit the PR:

* ~[ ] Has `CHANGELOG.md` been updated?~
* ~[ ] Are tests passing locally?~
* [x] Are the files formatted correctly?
* ~[ ] Did you add unit tests?~
* [X] Did you check UI tests on the sample app? They are not executed on CI.
* [X] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

There's a bug on Mac Catalyst and UIPicker isn't working on it. This PR adds an alternative implementation via `UIAlertController`.
image

## Misc

[AB#81428](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/81428)

### CODE REVIEW

File: Sasquatch/Sasquatch/ViewControllers/Cells/MSCustomPropertyTableViewCell.swift [hunk 1]
```diff
@@ -24,6 +24,8 @@ import UIKit
   private var typePickerView: MSEnumPicker<CustomPropertyType>?
   private var datePickerView: MSDatePicker?
 
+  public var viewController: UIViewController?
```
Comment (MatkovIvan) [inline Sasquatch/Sasquatch/ViewControllers/Cells/MSCustomPropertyTableViewCell.swift:27]: Should we add `#if targetEnvironment(macCatalyst)` here?

File: Sasquatch/Sasquatch/ViewControllers/MSEnumPicker.swift [hunk 2]
```diff
@@ -42,6 +55,29 @@ class MSEnumPicker<E: RawRepresentable & Equatable> : NSObject, UIPickerViewData
...
   
+  func showActionSheetPicker() {
+    let optionMenu = UIAlertController(title: nil, message: "Choose Option", preferredStyle: .alert)
+    let pickedValue = E(rawValue: self.textField.text!)!
+    for value in self.allValues {
+      
+      // Red style for picked option.
+      let style: UIAlertAction.Style = value.rawValue == pickedValue.rawValue ? .destructive : .default
+      let action = UIAlertAction(title: value.rawValue, style: style, handler: {
+        (_ action : UIAlertAction) -> Void in
```
Comment (MatkovIvan) [inline Sasquatch/Sasquatch/ViewControllers/MSEnumPicker.swift:66]: Is this formatted correctly?

File: Sasquatch/Sasquatch/ViewControllers/Cells/MSCustomPropertyTableViewCell.swift [hunk 3]
```diff
@@ -92,6 +96,18 @@ import UIKit
     prepareForReuse()
   }
   
+  override func didTransition(to state: UITableViewCell.StateMask) {
+    
+    // We need to pass the parent view controller to the picker
+    // to be able to show the alert on mac catalyst.
+    // We need to do it here, where we know the viewController is already set.
+#if targetEnvironment(macCatalyst)
```
Comment (MatkovIvan) [inline Sasquatch/Sasquatch/ViewControllers/Cells/MSCustomPropertyTableViewCell.swift:104]: Why platform specific logic is placed out of `MSEnumPicker`? Can we put all conditionals inside? if you need extra data there it probably makes sense to just pass it every time

File: Sasquatch/Sasquatch/ViewControllers/MSEnumPicker.swift (hunk 2 above)
Comment (annakocheshkova) [reply Sasquatch/Sasquatch/ViewControllers/MSEnumPicker.swift:66]: by `clang-format-changed-files`

**Pull request merged**